test/JDBC/expected/Test-sp_execute_postgresql.out

```diff
@@ -0,0 +1,231 @@
+
+
+
+-- Creates extension even if extension statement contains leading/trailing spaces by removing trailing spaces
+exec sp_execute_postgresql '     create extension pg_stat_statements      ';
+-- Create extension
+exec sp_execute_postgresql 'create extension if not exists pg_stat_statements';
+go
+select extname, extowner, extversion from pg_extension where extname = 'pg_stat_statements'
+go
+~~START~~
+varchar#!#int#!#text
+pg_stat_statements#!#16384#!#1.10
+~~END~~
+
+
+exec sp_execute_postgresql 'create extension if not exists pg_stat_statements';
+go
+
+-- psql
+SET pg_stat_statements.track = 'none';
+SELECT pg_stat_statements_reset();
+go
+~~START~~
+void
+
+~~END~~
+
+-- tsql
+-- for accessing the extension we need to give schema qualifier i.e. [public].pg_stat_statements
+SELECT dbid, toplevel, query, calls, rows, plans from [public].pg_stat_statements where queryid != 0 ORDER BY query COLLATE "C";
+go
+~~START~~
+int#!#bit#!#text#!#bigint#!#bigint#!#bigint
+~~END~~
+
+
+-- create extension without using 'if not exists' option will throw error
+exec sp_execute_postgresql 'create extension pg_stat_statements';
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: extension "pg_stat_statements" already exists)~~
+
+exec sp_execute_postgresql 'drop extension pg_stat_statements';
+go
+
+-- throw an error if user creates anything other than allowed extensions
+exec sp_execute_postgresql 'create extension pg_visibility';
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: 'pg_visibility' is not a valid name.)~~
+
+
+-- create extension with schema sys, public or by default(public), else throw error
+exec sp_execute_postgresql 'create extension pg_stat_statements with schema sys';
+go
+
+exec sp_execute_postgresql 'create extension pg_stat_statements with schema master_dbo';
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: 'master_dbo' is not a valid schema, only sys and public schema is allowed in Babelfish.)~~
+
+
+exec sp_execute_postgresql 'drop extension pg_stat_statements';
+go
+-- create extension with any previous version
+exec sp_execute_postgresql 'create extension pg_stat_statements with version "1.4"';
+go
+
+select extname, extowner, extversion from pg_extension where extname = 'pg_stat_statements'
+go
+~~START~~
+varchar#!#int#!#text
+pg_stat_statements#!#16384#!#1.4
+~~END~~
+
+
+-- cascade not supported yet, it will throw error
+exec sp_execute_postgresql 'create extension pg_stat_statements with cascade';
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: 'cascade' is not yet supported in Babelfish.)~~
+
+
+-- throw an error if more than one statement is passed
+exec sp_execute_postgresql 'create extension pg_stat_statements with schema sys; create extension pg_stat_statements';
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: Expected 1 statement but got 2 statements after parsing.)~~
+
+
+--alter extension version
+exec sp_execute_postgresql 'alter extension pg_stat_statements update to "1.10"';
```

Comment:
To verify this, we need to install a version lower than 1.10 and update it to 1.10.